recbole/model/general_recommender/addease.py

```diff
@@ -0,0 +1,124 @@
+
+r"""
+ADD-EASE
+################################################
+Reference:
+    Olivier Jeunen, et al. "Closed-Form Models for Collaborative Filtering with Side-Information".
+
+Reference code:
+    https://github.com/olivierjeunen/ease-side-info-recsys-2020/
+"""
+
+
+from recbole.utils.enum_type import ModelType, FeatureType
+import numpy as np
+import scipy.sparse as sp
+import torch
+
+from recbole.utils import InputType
+from recbole.model.abstract_recommender import GeneralRecommender
+
+from sklearn.preprocessing import MultiLabelBinarizer, OneHotEncoder
+
+
+def encode_categorical_item_features(dataset, selected_features):
+    item_features = dataset.get_item_feature()
+
+    mlb = MultiLabelBinarizer(sparse_output=True)
+    ohe = OneHotEncoder(sparse=True)
+
+    encoded_feats = []
+
+    for feat in selected_features:
+        t = dataset.field2type[feat]
+        feat_frame = item_features[feat].numpy()
+
+        if t == FeatureType.TOKEN:
+            encoded = ohe.fit_transform(feat_frame.reshape(-1, 1))
+            encoded_feats.append(encoded)
+        elif t == FeatureType.TOKEN_SEQ:
+            encoded = mlb.fit_transform(feat_frame)
+
+            # drop first column which corresponds to the padding 0; real categories start at 1
+            # convert to csc first?
+            encoded = encoded[:, 1:]
+            encoded_feats.append(encoded)
+        else:
+            raise Warning(
+                f'ADD-EASE only supports token or token_seq types. [{feat}] is of type [{t}].')
+
+    if not encoded_feats:
+        raise ValueError(
+            f'No valid token or token_seq features to include.')
+
+    return sp.hstack(encoded_feats).T.astype(np.float32)
+
+
+def ease_like(M, reg_weight):
+    # gram matrix
+    G = M.T @ M
+
+    # add reg to diagonal
+    G += reg_weight * sp.identity(G.shape[0])
+
+    # convert to dense because inverse will be dense
+    G = G.todense()
+
+    # invert. this takes most of the time
+    P = np.linalg.inv(G)
+    B = P / (-np.diag(P))
+    # zero out diag
+    np.fill_diagonal(B, 0.)
+
+    return B
+
+
+class ADDEASE(GeneralRecommender):
+    input_type = InputType.POINTWISE
+    type = ModelType.TRADITIONAL
+
+    def __init__(self, config, dataset):
+        super().__init__(config, dataset)
+
+        # need at least one param
+        self.dummy_param = torch.nn.Parameter(torch.zeros(1))
+
+        inter_matrix = dataset.inter_matrix(
+            form='csr').astype(np.float32)
+
+        item_feat_proportion = config['item_feat_proportion']
+        inter_reg_weight = config['inter_reg_weight']
+        item_reg_weight = config['item_reg_weight']
+        selected_features = config['selected_features']
+
+        tag_item_matrix = encode_categorical_item_features(
```

Comment:
To sovle the above issue, you can clip the tag_item_matrix by num_items.
`tag_item_matrix = tag_item_matrix[:, :self.num_items]`

Or just filter the items by config parameter.
`tem_inter_num_interval:  [1,inf)`